.github/workflows/unit_tests.yml

```
@@ -0,0 +1,49 @@
+name: unit tests
+
+on:
+  push: # run on every push or PR to any branch
+  pull_request:
+  schedule: # run automatically on main branch each Tuesday at 11am
+    - cron: "0 16 * * 2"
+
+jobs:
+  python-unit:
+    name: Python unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ["3.12"]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      # We base the python cache on the hash of all requirements files, so that
+      # if any change, the cache is invalidated.
+      - name: Cache pip
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('pyproject.toml') }}
+          restore-keys: |
+            pip-${{ hashFiles('pyproject.toml') }}
+            pip-
+
+      - name: Install package with dependencies
+        run: |
+          pip install -e .
+          pip install -e '.[dev]'
+          pip install codecov
+
+      - name: Generate PLY lextab and parsetab artifacts
+        run: python -c "import neuxml.xpath.core"
+
+      - name: Run pytest
+        run: pytest --cov=neuxml --cov-report=xml
+
+      - name: Upload test coverage to Codecov
+        uses: codecov/codecov-action@v3
```

This is what Codecov says the task should look like - not sure what the indentation should be, please adjust

```suggestion
- name: Upload coverage reports to Codecov
    uses: codecov/codecov-action@v5
    with:
      token: ${{ secrets.CODECOV_TOKEN }}
      slug: Princeton-CDH/neuxml
```

---

Once you add other python versions, let's pick a version and only run this once per build. I have an example of this somewhere if you want.